mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -1082,11 +1089,19 @@ def _patch_points_sampling(self, points, sem_mask, replace=None):
             if flag1 and flag2:
                 break
 
-        # random sample idx
-        if replace is None:
-            replace = (cur_sem_mask.shape[0] < self.num_points)
-        choices = np.random.choice(
-            np.where(cur_choice)[0], self.num_points, replace=replace)
+        # sample idx to `self.num_points`
+        if point_idxs.size >= self.num_points:
```

Comment:
The change of sampling strategy here does slightly alter the function behavior, but it's a better and more widely used one (in segmentation papers later than PointNet++). It will actually improve the performance.